kratos/includes/amatrix_interface.h

```diff
@@ -609,6 +609,19 @@ typename TExpression1Type::data_type inner_prod(
 
 }
 
+    template <typename TExpressionType, std::size_t TCategory>
+    typename TExpressionType::data_type norm_1(
+        AMatrix::MatrixExpression<TExpressionType, TCategory> const& TheExpression) {
+            using data_type = typename TExpressionType::data_type;
+            auto& the_expression = TheExpression;
```

Comment:
I don't know how it compiles like this....

It should be:
``` c++
auto& the_expression = TheExpression..expression();
```

Reply:
it also compiled with auto& the_expression = TheExpression; but I changed to your suggestion in the new commit